src/persist-client/src/batch.rs

```diff
@@ -989,22 +987,19 @@ impl<T: Timestamp + Codec64> BatchParts<T> {
                         });
 
                         // We can't collect stats if we failed to encode in a columnar format.
-                        let Ok(((key_col, val_col), stats)) = result else {
+                        let Ok((extended_cols, stats)) = result else {
                             tracing::error!(?result, "failed to encode in columnar format!");
                             break 'collect_stats None;
                         };
 
                         // Write a structured batch if the dyncfg is enabled and we're the stats
                         // override is not set.
                         if let BlobTraceUpdates::Row(record) = &updates.updates {
-                            if cfg.batch_columnar_format.is_structured()
-                                && !cfg.batch_columnar_stats_only_override
-                            {
-                                let record_ext = ColumnarRecordsStructuredExt {
-                                    key: key_col,
-                                    val: val_col,
-                                };
```

Comment:
Notes on why making k_s and v_s non-optional is cool:
- This is the only place that `ColumnarRecordsStructuredExt` is constructed outside of serialization code.
- `key_col` and `val_col` are always `Some` here! If `batch_format.is_structured()`, `encode_updates` returns `Some` for both... and if not, we never enter this block.